test/app-tap/datetime.test.lua

```diff
@@ -1830,6 +1830,565 @@ test:test("Parse strptime format", function(test)
+        },
+        {
+            buf = '23',
+            fmt = '%H',
+        },
+        -- %I
+        {
+            buf = '01',
+            fmt = '%I',
+        },
+        {
+            buf = '12',
+            fmt = '%I',
+        },
+        -- %j
+        {
+            buf = '001',
+            fmt = '%j',
+        },
+        {
+            buf = '31-12-2024 366',
+            fmt = '%d-%m-%Y %j',
+        },
+        -- %k
+        {
+            buf = ' 0',
+            fmt = '%k',
+        },
+        {
+            buf = '23',
+            fmt = '%k',
+        },
+        -- %l
+        {
+            buf = ' 1',
+            fmt = '%l',
+        },
+        {
+            buf = '12',
+            fmt = '%l',
+        },
+        -- %m
+        {
+            buf = '01',
+            fmt = '%m',
+        },
+        {
+            buf = '12',
+            fmt = '%m',
+        },
+        -- %M
+        {
+            buf = '00',
+            fmt = '%M',
+        },
+        {
+            buf = '59',
+            fmt = '%M',
+        },
+        -- %n
+        {
+            buf = '\n',
+            fmt = '%n',
+        },
+        -- %O, modifier, skipped.
+        -- %p
+        {
+            buf = 'PM',
+            fmt = '%p',
+        },
+        {
+            buf = 'AM',
+            fmt = '%p',
+        },
+        -- %r, same as "%I:%M:%S %p".
+        {
+            buf = '12:32:01 AM',
+            fmt = '%r',
+        },
+        {
+            buf = '12:32:01 PM',
+            fmt = '%r',
+        },
+        -- %R, same as "%H:%M".
+        {
+            buf = '18:20',
+            fmt = '%R',
+        },
+        -- %s
+        {
+            buf = '26-08-2024 1724630400',
+            fmt = '%d-%m-%Y %s',
+        },
+        -- %S
+        {
+            buf = '00',
+            fmt = '%S',
+        },
+        {
+            buf = '59',
+            fmt = '%S',
+        },
+        -- %t
+        {
+            buf = '26-08-2024 \t',
+            fmt = '%d-%m-%Y %t',
+        },
+        -- %T, same as "%H:%M:%S".
+        {
+            buf = '00:00:00',
+            fmt = '%T',
+        },
+        {
+            buf = '23:59:59',
+            fmt = '%T',
+        },
+        -- %u
+        -- Works for 4 ('Thursday') only when date is not
+        -- complete.
+        {
+            buf = '4',
+            fmt = '%u',
+        },
+        {
+            buf = '30-08-2024 5',
+            fmt = '%d-%m-%Y %u',
+        },
+        {
+            buf = '26-08-2024 1',
+            fmt = '%d-%m-%Y %u',
+        },
+        -- %U
+        {
+            buf = '01-01-2024 00',
+            fmt = '%d-%m-%Y %U',
+        },
+        {
+            buf = '31-12-2024 52', -- "00 to 53"?
+            fmt = '%d-%m-%Y %U',
+        },
+        -- %v, same as "%e-%b-%".
+        {
+            buf = ' 1-Jan-1970',
+            fmt = '%v',
+        },
+        -- %V, unsupported by strptime(3).
```

Comment:
Side note: I have the following in the manual:
```
   glibc notes
       For reasons of symmetry, glibc tries to support for strptime() the same format characters as for strftime(3).  (In most cases, the corresponding fields
       are parsed, but no field in tm is changed.)  This leads to

...

       %V     The  ISO  8601:1988  week number as a decimal number (1–53).  If the week (starting on Monday) containing 1 January has four or more days in the
              new year, then it is considered week 1.  Otherwise, it is the last week of the previous year, and the next week is week 1.
```

So, the value is parsed, may be it is better to add the corresponding test?

Reply:
We use `strptime` implementation derived from FreeBSD (see commit message).
There is no glibc note in FreeBSD manual page, see [^1].

[^1]: https://man.freebsd.org/cgi/man.cgi?strftime(3)